var/spack/repos/builtin/packages/libquo/package.py

```diff
@@ -26,11 +26,14 @@ class Libquo(AutotoolsPackage):
 
     depends_on("mpi")
 
-    depends_on("m4", when="@develop", type="build")
-    depends_on("autoconf", when="@develop", type="build")
-    depends_on("automake", when="@develop", type="build")
-    depends_on("libtool", when="@develop", type="build")
+    with when("@develop"):
+        depends_on("m4", type="build")
+        depends_on("autoconf", type="build")
+        depends_on("automake", type="build")
+        depends_on("libtool", type="build")
```

Comment:
> This doesn't break anything, but I wonder if it would be confusing to see those Autotools-related concerns being expressed in the package itself. 

I don't see anything that might be confusing. I would probably code something like:
```python
with when("@develop buildsystem=autotools"):
    depends_on("m4", type="build")
    depends_on("autoconf", type="build")
    depends_on("automake", type="build")
    depends_on("libtool", type="build")
```
I wouldn't code a separate helper class, mainly because not all the packages need all of the tools (I think usually it's libtool to be optional / depending on the project).

Reply:
Putting this in the most general terms: it feels like a place where buildsystem-specific logic creeps into the `Package` when our goal is to avoid that. We can leave this as is though: I'm not sure how else to handle it given that the subset of dependencies changes on a package-by-package basis.